json-fleece-aeson/json-fleece-aeson.cabal

```diff
@@ -41,7 +41,8 @@ library
     , base >=4.7 && <5
     , bytestring ==0.11.*
     , containers ==0.6.*
-    , json-fleece-core ==0.7.*
+    , json-fleece-core ==0.8.*
+    , json-fleece-openapi3 ==0.5.*
```

Comment:
I don't love how this added this dependency to a bunch of the packages. The instances need to live somewhere, so the alternatives were reversing this and adding a `json-fleece-aeson` (and all of the others) dependency to `json-fleece-openapi3`, or putting the `OpenApi3Validator` class in `json-fleece-core`. Neither of those felt right. We could use a cabal flag and the CPP to selectively add the dependency and instance instead.

Reply:
The swagger2 and openapi3 packages [aren't maintained very actively](https://github.com/biocad/openapi3/pulls), so I think it would be nice if we could avoid depending on them unconditionally.

Reply:
That's a good point. It's actually pretty terrible to have `openapi3` as a transitive dependency of `json-fleece-aeson` now that I think about it.

Reply:
I agree it would be nice to support using fleece without needing to depend on openapi. How bad does adding `json-fleece-aeson` and others as dependencies to `json-fleece-openapi3` seem? Ideally someone in theory could extend fleece with their own custom validation classes in a separate package right?

Reply:
> How bad does adding json-fleece-aeson and others as dependencies to json-fleece-openapi3 seem?

Given that json-fleece-aeson is a dependency of the generated code I don't think it would be bad. That being said, the class I originally called `OpenApi3Validator` should really just be something like `JsonSchemaValidator`, since it doesn't do anything OpenAPI-specific, and probably in a separate package.